oauth2-authorization-server/src/main/java/org/springframework/security/config/annotation/web/configuration/OAuth2AuthorizationServerConfiguration.java

```diff
@@ -47,12 +48,17 @@ public static void applyDefaultSecurity(HttpSecurity http) throws Exception {
 				new OAuth2AuthorizationServerConfigurer<>();
 		RequestMatcher endpointsMatcher = authorizationServerConfigurer
 				.getEndpointsMatcher();
+		RequestMatcher oidcClientRegistrationEndpointMatcher = authorizationServerConfigurer
+				.getOidcClientRegistrationEndpointMatcher();
 
 		http
 			.requestMatcher(endpointsMatcher)
 			.authorizeRequests(authorizeRequests ->
-				authorizeRequests.anyRequest().authenticated()
+						authorizeRequests.requestMatchers(oidcClientRegistrationEndpointMatcher)
+								.access("hasAuthority('SCOPE_client.create')")
+								.anyRequest().authenticated()
 			)
+			.oauth2ResourceServer(OAuth2ResourceServerConfigurer::jwt)
```

Comment:
This should only be applied if the registration endpoint is enabled

Reply:
hi @jgrandja . My previous comment was somehow lost. if we are creating our own AuthenticationProvider is the oauth2-resource-server still required? Or the intention is to use the ```JwtAuthenticationProvider``` to decode and validate the token, and the ```OidcClientRegistrationAuthenticationProvider``` will be called from the filter with a BearerAuthenticationToken?

Reply:
> is the oauth2-resource-server still required

Yes. It will simply decode and verify the token.

The `BearerTokenAuthenticationFilter` will ultimately set the `SecurityContext` with a `JwtAuthenticationToken` and the `OidcClientRegistrationAuthenticationProvider` will use the `JwtAuthenticationToken` to apply the authorization check. Basically, `BearerTokenAuthenticationFilter` will perform **authentication** via `JwtAuthenticationProvider` and `OidcClientRegistrationAuthenticationProvider` will perform **authorization**.

Reply:
> Also do you have a preferred approach on how to check if the registration endpoint is enabled at this stage?

Let's define a setting in `ProviderSettings`.

Reply:
Yes, I already added a new property in the ProviderSettings. I'm just not sure which is the best way to get access to  ProviderSettings from `applyDefaultSecurity` method

Reply:
`
http
				.requestMatcher(endpointsMatcher)
				.authorizeRequests(authorizeRequests ->
						authorizeRequests.anyRequest().authenticated()
				).csrf(csrf -> csrf.ignoringRequestMatchers(endpointsMatcher))
				.apply(authorizationServerConfigurer);

		if (authorizationServerConfigurer.oidcClientRegistrationEnabled()) {
			http.oauth2ResourceServer(OAuth2ResourceServerConfigurer::jwt);
		}
`
This is how I have it right now. `authorizationServerConfigurer.oidcClientRegistrationEnabled()` is using the ProviderSettings to read the setting.

Reply:
Let's leave it the way you have it now. But I think it will change. We might need to expose `ProviderSettings` in a different way so it's more accessible.

Reply:
I agree. It's not the easiest/best way to access `ProviderSettings`

Reply:
Fixed